From bf4dc41bd461c8eecc3f7944942b7af14eba0859 Mon Sep 17 00:00:00 2001
From: Michael Jeanson <mjeanson@efficios.com>
Date: Thu, 9 Jun 2016 12:16:30 -0400
Subject: [PATCH lttng-tools 1/4] Tests: Make warn_processes.sh more portable

Options to pgrep aren't standardised across coreutils
implementations, use a more common option.

Signed-off-by: Michael Jeanson <mjeanson@efficios.com>
---
 tests/Makefile.am                   |  2 +-
 tests/utils/Makefile.am             |  2 +-
 tests/utils/warn_lttng_processes.sh | 23 -----------------------
 tests/utils/warn_processes.sh       | 25 +++++++++++++++++++++++++
 4 files changed, 27 insertions(+), 25 deletions(-)
 delete mode 100755 tests/utils/warn_lttng_processes.sh
 create mode 100755 tests/utils/warn_processes.sh

diff --git a/tests/Makefile.am b/tests/Makefile.am
index 58caeb1..3600e99 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -5,7 +5,7 @@ if BUILD_TESTS
 SUBDIRS += . utils regression unit stress destructive
 if HAS_PGREP
 check-am:
-	$(top_srcdir)/tests/utils/warn_lttng_processes.sh $(PGREP)
+	$(top_srcdir)/tests/utils/warn_processes.sh $(PGREP)
 endif
 else
 	@echo "========================================="
diff --git a/tests/utils/Makefile.am b/tests/utils/Makefile.am
index bd79d20..9d327e0 100644
--- a/tests/utils/Makefile.am
+++ b/tests/utils/Makefile.am
@@ -1,6 +1,6 @@
 SUBDIRS = . tap testapp
 
-EXTRA_DIST = utils.sh test_utils.py babelstats.pl warn_lttng_processes.sh
+EXTRA_DIST = utils.sh test_utils.py babelstats.pl warn_processes.sh
 dist_noinst_SCRIPTS = utils.sh test_utils.py babelstats.pl
 noinst_LTLIBRARIES = libtestutils.la
 
diff --git a/tests/utils/warn_processes.sh b/tests/utils/warn_processes.sh
new file mode 100755
index 0000000..fee1d0f
--- /dev/null
+++ b/tests/utils/warn_processes.sh
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+PGREP="$1"
+
+if [ x$LTTNG_TOOLS_TESTS_DISABLE_WARN_LTTNG_PROCESSES == x1 ]; then
+	exit
+fi
+
+color_warn='\E[1;33m'
+color_reset='\E[0m'
+color_bold='\E[1m'
+
+lttng_processes="$("$PGREP" -l 'lttng|gen-ust-.+')"
+
+if [ $? -eq 0 ]; then
+	pids="$(cut -d ' ' -f 1 <<< "$lttng_processes" | tr '\n' ' ')"
+
+	echo -e "${color_warn}Warning: the following LTTng processes were detected running on the system:$color_reset"
+	echo
+	echo "$lttng_processes"
+	echo
+	echo -e "Here's how to kill them: ${color_bold}kill -9 $pids$color_reset"
+	echo -e "${color_warn}If you leave them alive, some tests could fail.$color_reset"
+	echo
+fi
-- 
2.7.4

